sound/pci/hda/patch_hdmi.c

```diff
@@ -1162,7 +1182,9 @@ static int hdmi_pcm_open_no_pin(struct hda_pcm_stream *hinfo,
 	if (pcm_idx < 0)
 		return -EINVAL;
 
-	err = hdmi_choose_cvt(codec, -1, &cvt_idx);
+	err = hdmi_choose_cvt_nopin(codec, pcm_idx, &cvt_idx);
+	if (err)
+		err = hdmi_choose_cvt(codec, -1, &cvt_idx);
```

Comment:
commit message nit-pick:
Fix the problem by prioritizing the converter with the same index as PCM

This isn't really prioritizing, you first try with the nopin version then fallback to the existing solution.